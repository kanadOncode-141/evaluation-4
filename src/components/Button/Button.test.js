import Button from "./Button";

import { cleanup, getByTestId, render } from "@testing-library/react"
import { ReactDOM } from "react";
import React from "react";
import renderer from 'react-test-renderer'

afterEach(cleanup)

describe('Button Components', () => { 
    it("Should render Properly", () =>{
        render(<Button name = "A B"/>);
    })
    
    it("Should render without crasing",()=>{
        const div = document.createElement('div')
        ReactDOM.render(<Button rightIcon="button"></Button>, div)
    })

    it("should render button correctly with righticon copy",()=>{
        const {getByTestId}= render( <Button rightIcon="copy"></Button>)
        expect( getByTestId('button') ).toHaveTextContent('copy')
    })

    it("should render without crasing",()=>{
        const div = document.createElement('div')
        ReactDOM.render( <Button leftIcon="button"></Button>,div)
      })

    it("should render button correctly with leftIcon copy",()=>{
        const {getByTestId}= render( <Button leftIcon="copy"></Button>)
        expect( getByTestId('button') ).toHaveTextContent('copy')
    })

    it("should render without crasingn", ()=>{
        const div = document.createElement('div')
        ReactDOM.render( <Button children="button"></Button>,div)
    })
    
    it("should render button correctly with children copy",()=>{
        const {getByTestId}= render( <Button children="copy"></Button>)
        expect( getByTestId('button') ).toHaveTextContent('copy')
    })

    it("should render button correctly with leftIcon add",()=>{
        const {getByTestId}= render( <Button label="add"></Button>)
        expect( getByTestId('button') ).toHaveTextContent('add')
    })

    it("matches snapshot", ()=>{
        const tree = renderer.create(<Button label="REMOVE"></Button>).toJSON()
        expect(tree).toMatchSnapshot()
    })
 });
