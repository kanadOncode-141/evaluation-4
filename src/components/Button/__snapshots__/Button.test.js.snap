// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Components matches snapshot 1`] = `
<button
  className="sc-bczRLJ kfNcSk"
  data-testid="button"
  disabled={false}
  onClick={[Function]}
  size="sm"
/>
`;
